src/python_testing/TC_AccessChecker.py

```diff
@@ -137,20 +141,38 @@ async def _run_read_access_test_for_cluster_privilege(self, endpoint_id, cluster
     async def _run_write_access_test_for_cluster_privilege(self, endpoint_id, cluster_id, cluster, xml_cluster: XmlCluster, privilege: Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum, wildcard_read):
         for attribute_id in checkable_attributes(cluster_id, cluster, xml_cluster):
             spec_requires = xml_cluster.attributes[attribute_id].write_access
-            if spec_requires == Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kUnknownEnumValue:
-                # not writeable
-                continue
+            is_optional_write = xml_cluster.attributes[attribute_id].write_optional
+
             attribute = Clusters.ClusterObjects.ALL_ATTRIBUTES[cluster_id][attribute_id]
             cluster_class = Clusters.ClusterObjects.ALL_CLUSTERS[cluster_id]
             location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, attribute_id=attribute_id)
             test_name = f'Write access checker - {privilege}'
+            logging.info(f"Testing attribute {attribute} on endpoint {endpoint_id}")
+            if attribute == Clusters.AccessControl.Attributes.Acl and privilege == Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kAdminister:
+                logging.info("Skipping ACL attribute check for admin privilege as this is known to be writeable and is being used for this test")
+                continue
 
-            print(f"Testing attribute {attribute} on endpoint {endpoint_id}")
             # Because we read everything with admin, we should have this in the wildcard read
             # This will only not work if we end up with write-only attributes. We do not currently have any of these.
             val = wildcard_read.attributes[endpoint_id][cluster_class][attribute]
+            if isinstance(val, list):
+                # Use an empty list for writes in case the list is large and does not fit
+                val = []
+
             resp = await self.TH2.WriteAttribute(nodeid=self.dut_node_id, attributes=[(endpoint_id, attribute(val))])
-            if operation_allowed(spec_requires, privilege):
+            if spec_requires == Clusters.AccessControl.Enums.AccessControlEntryPrivilegeEnum.kUnknownEnumValue:
+                # not writeable - expect an unsupported write response
+                # Global vars currently return the wrong error code - see #31448
```

Comment:
should we wait for #31454 to be merged and then not commit the workaround?

Reply:
oh, awesome. Yes.

Reply:
kk, updated to remove that workaround. Thanks.